File: lib/rubeus/extensions/javax/swing/table/readonlyable_table_model.rb
Rubeus::Swing.depend_on('TableModel')
Rubeus::Swing.depend_on('DelegatableTableModel')

module Rubeus::Extensions::Javax::Swing::Table
  module ReadonlyableTableModel
    def method_missing(method, *args, &block)
      self.source.send(method, *args, &block)
    end
  end
end

File: java/src/main/java/jp/rubybizcommons/rubeus/extensions/javax/swing/table/ReadonlyableTableModel.java
package jp.rubybizcommons.rubeus.extensions.javax.swing.table;

import javax.swing.table.TableModel;

public class ReadonlyableTableModel extends DelegatableTableModel {

	public ReadonlyableTableModel(TableModel source) {
		super(source);
	}

	private boolean readonly = false;

	public boolean isReadonly() {
		return readonly;
	}

	public void setReadonly(boolean readonly) {
		this.readonly = readonly;
	}

	public boolean isCellEditable(int rowIndex, int columnIndex) {
		return !isReadonly();
	}
}
